virtual_ecosystem/models/animals/animal_model.py

```diff
@@ -162,15 +168,61 @@ def from_config(
             "Information required to initialise the animal model successfully "
             "extracted."
         )
-        return cls(
+        model = cls(
             data=data,
             core_components=core_components,
             functional_groups=functional_groups,
             model_constants=model_constants,
         )
 
+        return model
+
     def setup(self) -> None:
-        """Function to set up the animal model."""
+        """Method to setup the animal model specific data variables."""
+
+        # animal respiration data variable
+        # the array should have one value for each animal community
+        n_communities = len(self.data.grid.cell_id)
+
+        # Initialize total_animal_respiration as a DataArray with a single dimension:
+        # cell_id
+        total_animal_respiration = DataArray(
+            zeros(
+                n_communities
+            ),  # Filled with zeros to start with no carbon production.
+            dims=["cell_id"],
+            coords={"cell_id": self.data.grid.cell_id},
+            name="total_animal_respiration",
+        )
```

Comment:
I'm not entirely sure because I don't know the internal structure of `cell_id`, but maybe you can use:
```suggestion
        # Initialize total_animal_respiration as a DataArray with a single dimension:
        # cell_id
        total_animal_respiration = xr.zeros_like(self.data.grid.cell_id).rename("total_animal_respiration")
```